### PR TITLE
BBH pipeline: find horizons in initial data, write masses & spins to file

### PR DESCRIPTION
**Pull request by nilsvu**
## Proposed changes

Automatically runs the horizon finder over initial data and writes horizon quantities to the reductions file. This makes important quantities like horizon masses and spins available to later steps in the pipeline.

### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `new feature` if appropriate.

### CODE REVIEW

File: support/Pipelines/Bbh/InitialData.py
Comment (nilsdeppe) [inline support/Pipelines/Bbh/InitialData.py]: I don't think we should expose the m_max option. Everything goes unstable if m_max < l_max. Probably the right thing to do is to just not expose the constructor from C++.

**Review by nilsdeppe — CHANGES_REQUESTED** [review]
I only glanced at this, so don't count it as a review...

File: src/IO/H5/Python/File.cpp
Comment (knelli2) [inline src/IO/H5/Python/File.cpp]: Hmm. It's not finding this file. Is it dependent on the pybind11 version?

File: support/Pipelines/Bbh/FindHorizon.py
Comment (knelli2) [inline support/Pipelines/Bbh/FindHorizon.py]: `xyz` instead of `012`? I think our tensors do `xyz` suffixes.

File: support/Pipelines/Bbh/FindHorizon.py
Comment (knelli2) [inline support/Pipelines/Bbh/FindHorizon.py:120]: I think we should keep all the horizon stuff in the same H5 file. It reduces the number of files on disk and that we have to keep track of. (like forwarding options through pipeline commands)

File: support/Pipelines/Bbh/InitialData.py
Comment (knelli2) [inline support/Pipelines/Bbh/InitialData.py]: I think you should try to find the subfile name from the input file like in #5957 so that we keep the ability to run the pipeline steps separately.

File: support/Pipelines/Bbh/InitialData.py
Comment (knelli2) [inline support/Pipelines/Bbh/InitialData.py]: I'm pretty sure it's guaranteed to always be the last observation, but is that true?

File: support/Pipelines/Bbh/InitialData.py
Comment (knelli2) [reply support/Pipelines/Bbh/InitialData.py]: Yeah probably not, but you're right, this is more an issue with the c++ binding. I think we can do this in a follow-up PR.

We should probably make LMax an option (and default it to something like 12). This won't matter as much once the horizon finder is adaptive, but for now would be nice to have the flexibility.

File: support/Pipelines/Bbh/InitialData.py
Comment (knelli2) [inline support/Pipelines/Bbh/InitialData.py]: If this is where the ID root find is going to go, thoughts on putting this (and the CLI command) in a separate python file just so things aren't too cluttered? Do we have another example of having two CLIs in the same file?

File: support/Pipelines/Bbh/InitialData.py
Comment (knelli2) [inline support/Pipelines/Bbh/InitialData.py]: Help text

File: support/Pipelines/Bbh/InitialData.yaml
Comment (knelli2) [inline support/Pipelines/Bbh/InitialData.yaml:14]: Does this now not have the ability to choose whether to continue with ringdown? I see in `InitialData.py` that `continue_with_ringdown` is always True.

**Review by knelli2 — CHANGES_REQUESTED** [review]

File: support/Pipelines/Bbh/InitialData.py
Comment (nilsvu) [reply support/Pipelines/Bbh/InitialData.py]: I switched the binding to the constructor without `m_max` and added an `l_max` option.

File: src/IO/H5/Python/File.cpp
Comment (nilsvu) [reply src/IO/H5/Python/File.cpp]: I moved this commit to #6090

File: support/Pipelines/Bbh/FindHorizon.py
Comment (nilsvu) [reply support/Pipelines/Bbh/FindHorizon.py:120]: The evolutions write to a separate surfaces file though. Don't you want to be consistent with that? For reading in the data I think you can't assume that both the masses+spins and the shape coeffs are in the same file anyway, because they are so different (doubles vs essentially volume data).

Edit: options that I can think of are:
- masses+spins in reductions file, coeffs in new surfaces file
- masses, spins, and coeffs in new horizons file that has no evolution analog

File: support/Pipelines/Bbh/InitialData.py
Comment (nilsvu) [reply support/Pipelines/Bbh/InitialData.py]: Yea that should be fine

File: support/Pipelines/Bbh/InitialData.py
Comment (nilsvu) [reply support/Pipelines/Bbh/InitialData.py]: Ok yea a separate file is probably better. I'll do that when squashing.

File: support/Pipelines/Bbh/InitialData.py
Comment (nilsvu) [reply support/Pipelines/Bbh/InitialData.py]: Positional arguments don't have a `help`, it's part of the main help text. Edited that.

File: support/Pipelines/Bbh/InitialData.yaml
Comment (nilsvu) [reply support/Pipelines/Bbh/InitialData.yaml:14]: If you run with `--evolve` it will also just try to continue with ringdown (which isn't working yet). That's the same behavior as before. I could disable that since it's not working yet?

The `start-inspiral` command still has the option whether or not to continue with ringdown.

File: support/Pipelines/Bbh/FindHorizon.py
Comment (knelli2) [reply support/Pipelines/Bbh/FindHorizon.py:120]: I like the fixup you did for the postprocess step just writing things into a single `Horizons.h5` file. In the actual horizon find CLI, having the option for separate files is fine.

**Review by knelli2 — COMMENTED** [review]
Fixups look good. You can squash. We probably have to wait until #6090 is merged first?

**Review by nilsdeppe — DISMISSED** [review]
I don't need to look at the rest. Thanks for making the change to the bindings!

**Comment by nilsvu** [comment]
@knelli2 no I decoupled this from #6090. This is ready when you are :)

**Review by knelli2 — APPROVED** [review]

**Pull request merged**
